### DISCUSSION TITLE
Another Video Scripting issue. Sanchez/xplanet.

### DISCUSSION THREAD

**Discussion by stonegrift** [JVital2013/vitality-goes#79] [Q&A] [answered]
My issue and trying to figure it out. when I run Sanchez.sh I get thrown an error from xplanet. On every file attempts to process. on line 87 it gets killed. Followed by a line 106 error showing my path to Sanchez and :cannot execute: required file not found.

I have no idea if it's a Sanchez error or Xplanet. Or if one is feeding into the other.  Being a bash scripting and RegEx neophyte I'm lost.  Anyone have a list of which files are required where in Sanchez and xplanet?  I can't help feeling like people are talking over my head.

**Comment by JVital2013** [comment]
How much memory does your system have? It's likely not actually getting killed on line 87 (might just be the shell bugging out and saying that) - line 87 is just the beginning of a for loop. Most likely xplanet is using too much RAM and it gets killed on lines 105, 130, or 172 (the xplanet lines in the current version of the script).

Try changing the `--geometry` parameters on these lines from `21600x10800` to `2160x1080`. If that works, try `4320x2160`, then `8640x4320`, etc until you find the cut-off.

**Comment by stonegrift** [comment]
The Pi 4 has 4 gigs of ram and a 512 gig memory card. I tried reducing the geometry to no effect. The actual error is this (if I can get it to copy right.) [2024-06-24 09:21:45] Creating GOES 16 Sanchez False Color Images
[2024-06-24 09:21:45] Creating GOES16_FD_sanchez_20240623T145022Z.jpg...
Sanchez.sh: line 87: 1849659 Killed                  xplanet -config "$(dirname "$(readlink -fm "$0")")/Resources/xplanet-$satelliteMonth" -body earth -projection rectangular -num_times 1 -geometry 2160x1080 -date $thisDate.$thisTime -output /tmp/underlay.jpg
Sanchez.sh: line 106: /home/Warren/Sanchez/Sanchez: cannot execute: required file not found

I looked in the /tmp file and there was no underlay.jpg left sitting in that directory.  So I don't think it's even getting that far.  But running the script, the first thing it does before the xplanet line is delete underlay.  If I understood what the variables were suppose to be I could try running the xplanet line solo to see if it throws any errors.

It still doesn't seem to throw any errors. I ran a single line of xplanet 

xplanet -config /var/www/html/config/Resources/xplanet-06 -body earth -projection rectangular -num_times 1 -geometry 21600x10800 -date 20240624.130000 -output /tmp/underlay.jpg -verbosity 4

The verbosity at 4 told me everything it was doing. At first it had problems finding the font and I fixed that. Then it started finding the topo map and the dnb_land_ocean_ice file. then it just says "killed."

Here's the xplanet response to that command.

Looking for /var/www/html/config/Resources/xplanet-06 ... found
value is BODY, returnString is default
value is BUMP_SCALE, returnString is 1
value is CLOUD_SSEC, returnString is false
value is CLOUD_THRESHOLD, returnString is 90
value is COLOR, returnString is 255,255,255
value is MAGNIFY, returnString is 1
value is ORBIT, returnString is -.5,.5,2
value is ORBIT_COLOR, returnString is 255,255,255
value is RANDOM_ORIGIN, returnString is true
value is RANDOM_TARGET, returnString is true
value is SHADE, returnString is 30
value is TWILIGHT, returnString is 6
value is BODY, returnString is earth
value is DAY_MAP, returnString is /var/www/html/config/Resources/world.topo.200406.3x21600x10800.jpg
value is NIGHT_MAP, returnString is /var/www/html/config/Resources/dnb_land_ocean_ice.2012.21600x10800_geo.jpg
Looking for FreeMonoBold.ttf ... not found
Looking for xplanet/FreeMonoBold.ttf ... not found
Looking for xplanet/fonts/FreeMonoBold.ttf ... not found
Looking for /home/Warren/.xplanet/FreeMonoBold.ttf ... not found
Looking for /home/Warren/.xplanet/fonts/FreeMonoBold.ttf ... not found
Looking for /usr/share/xplanet/FreeMonoBold.ttf ... not found
Looking for /usr/share/xplanet/fonts/FreeMonoBold.ttf ... found
Looking for /var/www/html/config/Resources/world.topo.200406.3x21600x10800.jpg ... found
Looking for /var/www/html/config/Resources/dnb_land_ocean_ice.2012.21600x10800_geo.jpg ... found
Killed

That response is different than when I run Sanchez  which responds with 

[2024-06-25 13:20:42] Creating GOES16_FD_sanchez_20240624T175020Z.jpg...
Sanchez.sh: line 87: 2531152 Killed                  xplanet -config "$(dirname "$(readlink -fm "$0")")/Resources/xplanet-$satelliteMonth" -body earth -projection rectangular -num_times 1 -geometry 21600x10800 -date $thisDate.$thisTime -output /tmp/underlay.jpg

It's like it's seeing the variable names rather than the data.  End result is both are failing.

**Comment by stonegrift** [comment, marked as answer]
I got Sanchez to work by going back and building sanchez from the source rather than using the deb file.
